plugins/@grouparoo/pardot/src/lib/export/exportProfiles.ts

```diff
@@ -0,0 +1,273 @@
+import {
+  buildBatchExports,
+  BatchExport,
+  exportProfilesInBatch,
+  BatchSyncMode,
+  BatchGroupMode,
+  BatchMethodGetClient,
+  BatchMethodFindAndSetDestinationIds,
+  BatchMethodDeleteByDestinationIds,
+  BatchMethodUpdateByDestinationIds,
+  BatchMethodCreateByForeignKeyAndSetDestinationIds,
+  BatchMethodAddToGroups,
+  BatchMethodNormalizeForeignKeyValue,
+  BatchMethodNormalizeGroupName,
+  BatchMethodRemoveFromGroups,
+} from "@grouparoo/app-templates/dist/destination/batch";
+import { ExportProfilesPluginMethod } from "@grouparoo/core";
+import PardotClient from "../client";
+import { connect } from "../connect";
+import { PardotCacheData, getListId } from "./listMethods";
+
+interface PardotData {
+  cacheData: PardotCacheData;
+}
+
+// return an object that you can connect with
+const getClient: BatchMethodGetClient = async ({ config }) => {
+  return connect(config.appOptions);
+};
+
+// fetch using the keys to set destinationId and result on BatchExports
+// use the getByForeignKey to lookup results
+const findAndSetDestinationIds: BatchMethodFindAndSetDestinationIds = async ({
+  client,
+  foreignKeys,
+  getByForeignKey,
+}) => {
+  for (const email of foreignKeys) {
+    const prospect = await client.getProspectByEmail(email);
```

Comment:
probably `try` around this and set error so a certain users doesn't mess everyone else up.